src/Controller/DefaultController.php

```diff
@@ -136,24 +136,32 @@ public function admin()
     /**
      * Get the sitemap.xml containing all dataset urls.
      *
+     * @param FundingOrgFilter $fundingOrgFilter The funding organization filter utility.
+     *
      * @Route("/sitemap.xml", name="pelagos_sitemap")
      *
      * @return StreamedResponse
      */
-    public function showSiteMapXml()
+    public function showSiteMapXml(FundingOrgFilter $fundingOrgFilter)
     {
-        $response = new StreamedResponse(function () {
+        $criteria = array(
+            'availabilityStatus' =>
+            array(
+                DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE,
+                DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED,
+            )
+        );
 
-            $datasets = $this->getDoctrine()->getRepository(Dataset::class)->findBy(
-                array(
-                    'availabilityStatus' =>
-                    array(
-                        DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE,
-                        DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED,
-                    )
-                )
-            );
+        if ($fundingOrgFilter->isActive()) {
+            $criteria = array_merge($criteria, array(
+                'researchGroup' =>
+                $fundingOrgFilter->getResearchGroupsIdArray()
+            ));
+        }
+
+        $datasets = $this->getDoctrine()->getRepository(Dataset::class)->findBy($criteria);
 
+        $response = new StreamedResponse(function () use (&$datasets) {
```

Comment:
Do we need to pass it by reference here?

Reply:
No you don't have to, but otherwise a copy will be made. Not that is matters really.

Reply:
I think objects are passed by references by default, I don't think we need the "&" here